nodestream_akamai/gtm/gtm.py

```diff
@@ -10,35 +11,79 @@ def __init__(self, **akamai_client_kwargs) -> None:
         self.client = AkamaiGtmClient(**akamai_client_kwargs)
         self.logger = logging.getLogger(self.__class__.__name__)
 
+    def get_target_type(self, target) -> str:
+        parsed_target = "endpoint"
+        try:
+            parsed_ip = ip_address(target)
+            if type(parsed_ip) is IPv4Address:
+                parsed_target = "ipv4"
+            elif type(parsed_ip) is IPv6Address:
+                parsed_target = "ipv6"
+        except ValueError:
+            pass
+        return parsed_target
+
     async def extract_records(self):
+        # We only extract linked ipv4, ipv6 and endpoint types from these property types
+        parsed_types = [
+            "weighted-round-robin",
+            "weighted-hashed",
+            "weighted-round-robin-load-feedback",
+            "ranked-failover",
+            "failover",
+        ]
         for domain in self.client.list_gtm_domains():
             try:
                 raw_domain = self.client.get_gtm_domain(domain["name"])
-                deeplink_prefix = "https://control.akamai.com/apps/gtm/#/domains/"
-                deeplink = f'{deeplink_prefix}{domain["name"]}/properties/list'
-                properties = []
-                for property in raw_domain["properties"]:
-                    fqdn = property["name"] + "." + raw_domain["name"]
-                    servers = []
+            except Exception as err:
+                self.logger.error(f"Failed to get gtm domain: {domain['name']}")
+                self.logger.error(err)
+            deeplink_prefix = "https://control.akamai.com/apps/gtm/#/domains/"
+            deeplink = f'{deeplink_prefix}{domain["name"]}/properties/list'
+            properties = []
+            for property in raw_domain["properties"]:
+                fqdn = property["name"] + "." + raw_domain["name"]
```

Comment:
Documenting my general request to double-check map gets to make sure that the null/missing case is handled.

Reply:
Good callout, but in this case these fields are always present in the API response. We could handle this, but it seems like overkill, and they are required so an error is not a terrible outcome if they are missing. Happy to accommodate, though